src/imageio/storage/email.c

```diff
@@ -202,26 +206,37 @@ void finalize_store(dt_imageio_module_storage_t *self, dt_imageio_module_data_t
     gchar *filename = g_path_get_basename(attachment->file);
     const dt_image_t *img = dt_image_cache_get(darktable.image_cache, attachment->imgid, 'r');
     dt_image_print_exif(img, exif, sizeof(exif));
+
     g_snprintf(body + strlen(body), sizeof(body) - strlen(body), imageBodyFormat, filename, exif);
     g_free(filename);
 
-    if(*attachments)
-      g_snprintf(attachments + strlen(attachments), sizeof(attachments) - strlen(attachments), "%s",
-                 attachmentSeparator);
+    argv[n]   = g_strdup("--attach");
+    argv[n+1] = g_strdup(attachment->file);
+    n += 2;
 
-    g_snprintf(attachments + strlen(attachments), sizeof(attachments) - strlen(attachments), attachmentFormat,
-               attachment->file);
     // Free attachment item and remove
     dt_image_cache_read_release(darktable.image_cache, img);
     g_free(d->images->data);
```

Comment:
While you are at it, I have the feeling that we are leaking `attachment->file`. Either free that or use it directly in `argv`.